lib/jekyll/url.rb

```diff
@@ -94,7 +94,8 @@ def possible_keys(key)
 
     def generate_url_from_drop(template)
       template.gsub(%r!:([a-z_]+)!) do |match|
-        pool = possible_keys(match.sub(":", ""))
+        name = Regexp.last_match(1)
+        pool = name.end_with?("_") ? [name, name.chomp!("_")] : [name]
```

Comment:
Notes:
- `Regexp.last_match(1)` here is equivalent to `match.sub(":", "")` but without the extra allocation.
- The `#possible_keys` public method does not mutate the argument `key` but this inlined version modifies the string in-place.

Reply:
Sadly, It is not a private method. So we'll have to bear the tech debt until v5.0